test/NuGet.Core.Tests/NuGet.Commands.Test/TransitiveNoWarnUtilsTests.cs

```diff
@@ -928,5 +929,184 @@ public void DependencyNodeEqualityFails_NodesHaveDifferentPackageIdsInPackageSpe
             // Assert
             seen.Count.Should().Be(2);
         }
+
+        [Theory]
+        [InlineData(null, null, true)]
+        [InlineData(null, "", true)]
+        [InlineData(null, "NU1605", false)]
+        [InlineData("", null, true)]
+        [InlineData("", "", true)]
+        [InlineData("", "NU1605", false)]
+        [InlineData("NU1605", null, true)]
+        [InlineData("NU1605", "", true)]
+        [InlineData("NU1605", "NU1605", true)]
+        [InlineData("NU1605", "NU1604", false)]
+        public void NodeWarningPropertiesIsSubSetOf_ProjectWideNoWarns(
+            string firstNoWarn,
+            string secondNoWarn,
+            bool expected)
+        {
+            // Arrange
+            var first = new TransitiveNoWarnUtils.NodeWarningProperties(
+                firstNoWarn == null ? null : MSBuildStringUtility.GetNuGetLogCodes(firstNoWarn).ToHashSet(),
+                null);
+            var second = new TransitiveNoWarnUtils.NodeWarningProperties(
+                secondNoWarn == null ? null : MSBuildStringUtility.GetNuGetLogCodes(secondNoWarn).ToHashSet(),
+                null);
+
+            // Act
+            var result = first.IsSubSetOf(second);
+
+            // Assert
+            result.Should().Be(expected);
+        }
+
```

Comment:
Can we add a test case that assets similar log codes but different ids and there are with multiple ids in each.